cli/src/main/java/org/jboss/as/cli/handlers/module/ASModuleHandler.java

```diff
@@ -707,4 +716,38 @@ public static void copy(final File source, final File target) throws CommandLine
         }
     }
 
-}
+    private String getSchemaVersion(CommandContext ctx, boolean hasSlot) {
+        if (detectedSchemaVersion == null) {
+            MXParser parser = new MXParser();
+            final File wfCommonModule = new File(modulesDir, "system.layers.base.org.wildfly.common.main.".replace('.', File.separatorChar) + "module.xml");
+            try {
+                parser.setFeature(FEATURE_PROCESS_NAMESPACES, true);
+                parser.setInput(new FileInputStream(wfCommonModule), null);
+                parser.nextTag();
+            } catch (Exception ignore) {
+            }
+            detectedSchemaVersion = parser.getNamespace();
+        }
+
+        if (hasSlot && detectedSchemaVersion != null) {
```

Comment:
I think that we have the opportunity to create a "requiresSlot" function that would try to discover the current version. This function could be used in the slot parameter canAppearNext function (You can check in CLI code, we have a lot of examples) to only display the --slot argument (when pressing TAB) if the server supports it (so version <= 1.5). If any error occurs trying to discover the version, then we should fallback to use slot, the safer.